### PR TITLE
feat: add a virtual-passkey helper to @zitadel/testing

### PR DESCRIPTION
**Pull request by fforootd**
## Summary

Phase 5 of the test-kit track: **passkey testing lands in `@zitadel/testing`.** The kit was password-only while passkeys are the product's flagship flow — a passkey-first customer couldn't test their real auth path with it. This PR extracts the consumer journey's CI-proven CDP virtual-authenticator into the kit as public API and dogfoods it in three consumers. New public surface on a published package, hence `feat:` and a `minor` changeset.

- **`enableVirtualPasskey(page)`** (`packages/testing/src/passkey.ts`) + on-demand **`passkey` fixture**: attaches a virtual platform authenticator via CDP (ctap2, internal transport, discoverable credentials, user verified, automatic presence — exactly the profile the journey has run in CI), so real passkey registration and login ceremonies complete headlessly. The returned handle exposes `credentialCount()` (assert credential reuse via `expect.poll`) and `dispose()` — teardown the journey's private version never had. Clear error on non-Chromium projects (no CDP WebAuthn domain elsewhere).
- **Demo dogfood** (`apps/demo-next-e2e/src-real/passkey.spec.ts`): full round-trip on the existing e2e-real instance — fresh identity registers with a passkey at the registration-choice step of the *default password-first flow* (no second preset boot needed), signs out by clearing cookies **on the same page** (the credential lives in the page-bound authenticator), signs back in with the passkey, credential count stays 1. Shared widget-interaction helpers extracted to `src-real/flow-actions.ts` (were private copies in `registration.spec.ts`).
- **Journey improvements** (the dogfooding ask):
  - The **testkit consumer fixture** gains `fixtures/testkit/zitadel-e2e/passkey.spec.ts` — the same round-trip against the generated app, importing `enableVirtualPasskey` from the **registry-installed** kit: the customer-configuration proof now covers passkeys.
  - The **product journey** (`user-journey.spec.ts`) drops its private `enableVirtualAuthenticator`/`expectVirtualCredentialCount` (~35 lines) for the kit import — the packaged-product journey now consumes the kit helper on every CI run, both presets. Wiring: `@zitadel/testing` devDep + moon `testing:build` deps on the e2e tasks (automation → tool, same layering as demo-next-e2e).
- **README**: "Test passkey flows" section with the three inherent constraints (Chromium-only; page-bound authenticator — same-page round-trip, sign out via cookie clearing; `localhost` not `127.0.0.1` — WebAuthn rejects IP RP IDs) and the preset guidance (default flow offers passkey at the choice step; `passkey-first` for one-tap). Roadmap: ceremonies covered; server-side passkey *seeding* stays parked.

## Validation

- `moon run testing:build testing:test testing:typecheck testing:lint` — 61 unit tests / 11 files (new suite locks the exact CDP call sequence and authenticator options, `credentialCount`, dispose semantics incl. closed-target tolerance, and the Chromium error).
- `moon run demo-next-e2e:e2e-real` — **5/5** (new passkey round-trip 3.2s; suite 25.1s, 2 workers, one instance).
- `env -u CI moon run cli-journey-e2e:e2e-testkit` — **2/2** consumer specs (passkey 1.7s) with the kit installed from the journey registry.
- `env -u CI node .../run-local.mjs --framework next` — **3/3** product journey (password + passkey-only) through the kit-imported helper.
- `env -u CI node .../run-local.mjs --framework next --preset passkey-first` — **2/2 (7.0s)** (fallback registration + one-tap discoverable login).
- `moon run console-e2e:e2e-real` — **12/13**: the one failure is #673's new "renders exactly the properties the API's schema defines" spec, which fails **identically on pristine main** on this machine (reproduced 3/3 incl. clean tree @ `5253a1a4`; green in #673's CI) — pre-existing, environment-shaped, and fixed by #695 (readiness gate in `openDrawer`), not caused by this branch. With a readiness wait the lane runs 13/13 here. Every consumer-regression spec relevant to the kit passes.
- `moon run cli-journey-e2e:lint cli-journey-e2e:test` — journey scripts unaffected.

## Release notes / changeset

- `.changeset/testing-virtual-passkey.md` (`minor` for `@zitadel/testing`) — ships with the next alpha on the train.

## Notes

- Scope lines held: `seed.session()` stays password-only (a passkey step still fails loudly with the step name); server-side passkey seeding stays parked until server support exists; no multi-instance `withZitadel` (not needed — the default flow covers registration-choice passkeys).
- Remaining roadmap after this: OTP/email capture (server-gated), vitest entry (second backend consumer), project-per-worker (API-level suites), SQLite flip checklist (#665).

### CODE REVIEW

File: packages/testing/src/passkey.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    async dispose() {
+      // Best-effort: the page (and with it the CDP target) may already be
+      // gone when teardown runs after a failed test; never mask that failure.
+      try {
+        await client.send("WebAuthn.removeVirtualAuthenticator", { authenticatorId });
+        await client.detach();
+      } catch {
+        // ignore
+      }
```
[the remark this answers is not in the document]
Comment (fforootd) [reply packages/testing/src/passkey.ts:79]: Fixed in 22200f5f: removal and detach are separated — `detach()` now runs in a `finally` with its own swallow, so the session never outlives a failed removal, and the unit test asserts the detach still happens when removal throws.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: apps/demo-next-e2e/src-real/flow-actions.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  actionNames: string[],
+): Promise<void> {
+  const target = actionCandidates(page, name, actionNames).find(Boolean);
+  for (const candidate of actionCandidates(page, name, actionNames)) {
+    if (await candidate.first().isVisible({ timeout: 1_000 }).catch(() => false)) {
+      await candidate.first().click();
+      return;
+    }
+  }
+  await (target as Locator).first().click();
```
[the remark this answers is not in the document]
Comment (fforootd) [reply apps/demo-next-e2e/src-real/flow-actions.ts]: Fixed in 22200f5f: candidates are computed once and the fallback is an auto-waiting click on `candidates[0]` — the double locator creation and the cast are gone.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
